src/components/generic/InputField/InputField.jsx

```diff
@@ -49,6 +53,8 @@ export default function InputField({
   const inputRef = useRef();
 
   const handleChange = (event) => onChange(event.target.value);
+  const handleIncrement = () => value < max && onChange(Number(value) + 1);
```

Comment:
if the current value is 9.5 and max is 10, pressing up will go to 10.5

call onChange(Math.max(max, Number(value) + 1))